src/internal.c

```diff
@@ -5330,8 +5330,10 @@ static INLINE int GrowOutputBuffer(WOLFSSL* ssl, int size)
 #if WOLFSSL_GENERAL_ALIGNMENT > 0
     byte  hdrSz = ssl->options.dtls ? DTLS_RECORD_HEADER_SZ :
                                       RECORD_HEADER_SZ;
-#endif
+    byte align = WOLFSSL_GENERAL_ALIGNMENT;
+#else
     const byte align = WOLFSSL_GENERAL_ALIGNMENT;
```

Comment:
commit title says to only use align when WOLFSSL_GENERAL_ALIGNMENT is > 0.  But align here is set to WOLFSSL_GENERAL_ALIGNMENT and then used as an offset in XMALLOC.  In the zero case this is fine, in the negative case (which I'm guessing is the issue), not so good.